routing/base/astar_algorithm.hpp

```diff
@@ -231,25 +229,17 @@ class AStarAlgorithm
       return bestDistance.at(queue.top().vertex);
     }
 
-    // p_f(v) = 0.5*(π_f(v) - π_r(v)) + 0.5*π_r(t)
-    // p_r(v) = 0.5*(π_r(v) - π_f(v)) + 0.5*π_f(s)
+    // p_f(v) = 0.5*(π_f(v) - π_r(v))
+    // p_r(v) = 0.5*(π_r(v) - π_f(v))
     // p_r(v) + p_f(v) = const. Note: this condition is called consistence.
     Weight ConsistentHeuristic(Vertex const & v) const
     {
       auto const piF = graph.HeuristicCostEstimate(v, finalVertex);
       auto const piR = graph.HeuristicCostEstimate(v, startVertex);
       if (forward)
-      {
-        /// @todo careful: with this "return" here and below in the Backward case
-        /// the heuristic becomes inconsistent but still seems to work.
-        /// return HeuristicCostEstimate(v, finalVertex);
```

Comment:
The idea behind the comment was, I think, the following.
Making the proof work is quite a lot of work in the bidirectional case. One needs a way to reconcile the forward and backward heuristic functions. This is usually (since Ikeda, 1994) done by introducing the notion of consistence. Two arbitrary feasible functions are not necessarily consistent but it can be shown that some kind of averaging works, and that's what we use.
However, we could not find a counterexample to the much simpler solution: just use the heuristic function as is! The proof stops to work (or maybe it does with our choice of the heuristic function, we didn't try to check) but we could not find a counterexample. Moreover, on the current set of the routing integration tests only one (RussiaTaganrogSyzranov10k3ToLazo5k2) fails if we uncomment the line but the failure does not seem to result from the uncommenting, and rather from the wrong calculation of the path length in the test suite (it uses the reduced path lengths and not the real ones).
This is the reason for the todo: we either need a proof or a counterexample, and we don't have either.